src/java.se/share/data/jdwp/jdwp.spec

```diff
@@ -1981,8 +1981,8 @@ JDWP "Java(tm) Debug Wire Protocol"
     (Command CurrentContendedMonitor=9
         "Returns the object, if any, for which this thread is waiting. The "
         "thread may be waiting to enter a monitor, or it may be waiting, via "
-        "the java.lang.Object.wait method, for another thread to invoke the "
-        "notify method. "
+        "the java.lang.Object.wait method, to re-enter a monitor after being "
+        "notified by another thread. "
```

Comment:
Interrupted or timeout in Object.wait are other reasons to be waiting to reenter. I'm wondering if this should be spelled out here. I see the JVMTI spec uses the phrase "regain" to cover the scenarios where a thread is waiting to reenter.

In any case, I think the wording could be a smoother as reenter case isn't very clear. Here's an attempt as smoothing out the wording but it may be more than you want to include:

"The thread may be waiting to enter the object's monitor, in the monitor's wait set and waiting in Object.wait, or in Object.wait waiting to reenter the monitor after being notified, interrupted, or timeout."